### PR TITLE
Give a better error message if the HS doesn't support msisdn registeration

### PR DESCRIPTION
**Pull request by dbkr**
As commented, this isn't really the best way to do this, it just happens to be the only way it can fail right now, and doing it properly is a reasonable chunk of work that it doesn't necessarily make sense to be doing right now. Reckon this is OK for now?

### CODE REVIEW

**Comment by dbkr** [comment]
Have changed to emit a better error and added commit to https://github.com/matrix-org/matrix-react-sdk/pull/750 to handle it.

File: src/interactive-auth.js [hunk 1]
```diff
@@ -379,7 +379,16 @@ InteractiveAuth.prototype = {
                 return flow;
             }
         }
-        return null;
+        // XXX: This happens to be the only way this can fail right now, but
+        // in general this is not really going to be an accurate error message.
+        // Ideally this would signal what inputs could be removed such that a matching
+        // flow could be found.
+        const err = new Error("No appropriate authentication flow found");
+        err.required_stages = [];
```
Comment (richvdh) [inline src/interactive-auth.js]: I think if we're going to expect apps to catch this, we should give it a `name`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/name. `AuthenticationFlowNotFoundError` or something.

Can you also document the function(s) that will throw it?

File: src/interactive-auth.js [hunk 2]
```diff
@@ -379,7 +379,16 @@ InteractiveAuth.prototype = {
                 return flow;
             }
         }
-        return null;
+        // XXX: This happens to be the only way this can fail right now, but
+        // in general this is not really going to be an accurate error message.
```
Comment (richvdh) [inline src/interactive-auth.js]: It seems an accurate error message to me?

Comment (dbkr) [inline src/interactive-auth.js]: oops, updated

File: src/interactive-auth.js (hunk 1 above)
Comment (dbkr) [inline src/interactive-auth.js]: done

File: src/interactive-auth.js [hunk 3]
```diff
@@ -109,21 +109,25 @@ InteractiveAuth.prototype = {
     /**
      * begin the authentication process.
      *
-     * @return {module:client.Promise}  which resolves to the response on success,
-     * or rejects with the error on failure.
+     * @return {module:client.Promise} which resolves to the response on success,
+     * or rejects with the error on failure. Rejects with NoAuthFlowFoundError} if
```
Comment (richvdh) [inline src/interactive-auth.js]: }

**Pull request merged**